### PR TITLE
Ch/feature/apicalls

### LINKED ISSUE #6: As a user, I will see error messages
_As a user, when the server is down, I will see an appropriate error message._

Details:
- Within the body or main section of the webpage, under the nav bar, the user will see an error message detailing the website's issue

Acceptance Criteria:
_Scenario: a user cannot access the webpage._
The user sees the error message.

### LINKED ISSUE #7: As a user, I will see movies from database
_As a user, when I go to the homepage, I will see movies rendered from the API database._

Details:
- the hard coded data file will be replaced with the API data from a fetch

Acceptance Criteria:
_Scenario: a user on the homepage and the new movies are displayed._
All movies displayed are now from the API. The hard coded file has been deleted.

### PR DESCRIPTION
**Pull request by CaliHam**
### What is the feature?
this PR will close #7 
users can now see movies from the database and an error if no database is found

### How you implemented the solution?
used useEffect to call fetch function in apiCalls and fetch all movies from the database
used useState to configure error handling

### Does it impact any other area of the project?
deleted mock data since we are only using api data

### Any changes in the UI (Screenshots)?
example of error handling:
Screenshot 2023-07-07 at 3 45 42 PM

### How to test the feature (A test scenario or any new setup)?
npm start in terminal

### CODE REVIEW

**Review by sakisandrac — APPROVED** [review]
- Movies are rendering from the database as expected.
- When we simulate a 400 error, the error is displayed to the user on the homepage.
- Not sure if this will work for a 500 error as well but it should? Maybe we can somehow test this later.

**Comment by hfaerber** [comment]
> * Movies are rendering from the database as expected.
> * When we simulate a 400 error, the error is displayed to the user on the homepage.
> * Not sure if this will work for a 500 error as well but it should? Maybe we can somehow test this later.

You will be able to test this using Cypress.  You'll stub the fetch request and give a 500 level stubbed response to see how your application handles it.

File: src/apiCalls/apiCalls.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const getMovies = async () => {
```
Comment (hfaerber) [inline src/apiCalls/apiCalls.js:1]: Nice job declaring this in an apiCalls file.
Nice use of async/await - make sure you both understand and can explain how async/await is working here.

File: src/apiCalls/apiCalls.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    let response = await fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
+    if(!response.ok) {
+        console.log(response.statusText)
+        throw new Error(response.statusText)
```
Comment (hfaerber) [inline src/apiCalls/apiCalls.js:5]: Nice job checking for an ok response and throwing an Error here

File: src/components/App/App.js [hunk 3]
```diff
@@ -1,16 +1,25 @@
...
-  const [movies, setMovies] = useState(movieData.movies)
+  const [movies, setMovies] = useState([])
+  const [error, setError] = useState({isError:false, message: ''})
+
+  useEffect(() => {
+    getMovies().then(data => {
+      setError({isError:false, message: ''})
+      setMovies(data.movies)
+    })
+    .catch(err => setError({isError:true, message: err}))
```
Comment (hfaerber) [inline src/components/App/App.js:16]: Great use of the catch block here!

File: src/components/Homepage/Homepage.js [hunk 4]
```diff
@@ -3,7 +3,7 @@ import MoviePoster from '../MoviePoster/MoviePoster';
 import { useState } from 'react';
 import { MovieDetails } from '../MovieDetails/MovieDetails';
 
-const Homepage = ({ movies }) => {
+const Homepage = ({ movies, error }) => {
```
Comment (hfaerber) [inline src/components/Homepage/Homepage.js:6]: Nice use of destructuring here

File: src/components/Homepage/Homepage.js [hunk 5]
```diff
@@ -28,6 +28,7 @@ const Homepage = ({ movies }) => {
    return (
     <main className='main-container'>
         <h1 className='main-header'>Rancid Tomatillos</h1>
+        {error.isError && <p className='error'>{`Sorry! ${error.message}. Please try again later.`}</p>}
```
Comment (hfaerber) [inline src/components/Homepage/Homepage.js:31]: Good use of && operator

**Review by hfaerber — COMMENTED** [review]
Really nice work here - keep this up!  I've left a few comments for your review.

**Pull request merged**